### PR TITLE
+ command like arg: --config-file [./config.json]

### PR DESCRIPTION
**Pull request by ericprud**
This added arg allows you to override "`./config.json`". It's kinda lame in that it doesn't actually use the default 'cause config.json is read before `require('commander').parse(process.argv)` is run so you have to include the default a la:
```
let configFile = argv['configFile'] || './config.json'
```

### CODE REVIEW

File: bin/lib/start.js [hunk 1]
```diff
@@ -24,8 +24,9 @@ module.exports = function (program, server) {
 
   start.action((opts) => {
     let argv = extend({}, opts, { version: program.version() })
+    let configFile = argv['configFile'] || './config.json'
```
Comment (melvincarvalho) [inline bin/lib/start.js]: Might want to use path.join here with __dirname is that seems to be consistent with much of the code base

e.g. https://github.com/solid/node-solid-server/blob/dea121afbb92dd8e9ef8929d1d09246cd5ba5180/test/integration/capability-discovery-test.js#L12

**Comment by melvincarvalho** [comment]
Generally, looks good!

I think also solves an issue raised in 

https://github.com/solid/node-solid-server/issues/436

File: bin/lib/start.js (hunk 1 above)
Comment (ericprud) [reply bin/lib/start.js]: I'm happy to do that; was just making a minimal change to master. Does that buy us anything over just a relative path to config.json? I made pretty much the reverse change in shex.js when I made it able to load examples from a URL I suspect the only change is for node-solid-server is in debugging output.

Comment (melvincarvalho) [reply bin/lib/start.js]: No, looks good, happy to merge!

**Comment by dmitrizagidulin** [comment]
re relative path - we constantly run into trouble with relative paths with people who install solid server globally (npm i -g)

**Comment by melvincarvalho** [comment]
@dmitrizagidulin good point!  However the defaults for data and db are already ./data and ./.db respectively, right?

So ./config as a default seems consistent, for a quick change.  If I got that right.

We could perhaps cleanup the config a bit more later, e.g. in line with https://github.com/solid/node-solid-server/issues/436, alphabetizing the help etc.

**Comment by ericprud** [comment]
Not necessarily critical path for this PR, but what *is* the right behavior for servers installed with `-g`? Should their be platform-specific detection of whether it's installed in a user vs. non-user directory and in the latter case default to `/usr/local/etc` (and some equiv on windows)? This affects both install (defaults) and runtime (where to look for config.json).

**Comment by melvincarvalho** [comment]
@ericprud would you mind checking that the http options still works after this patch?

there was an issue raised : https://gitter.im/solid/node-solid-server?at=5ab044e8f3f6d24c68934a79

**Pull request merged**